### ISSUE TITLE
API Limitations

### ISSUE THREAD

**Issue by nil0x42** [zadam/trilium#685]
Hi ! I just started using trilium and i'm lovin' it.
Therefore, there seem to be some limitation in the API (but i am a javascript noob, so i might miss some information).

Let's take a look at the following `js` code:

```js
/*
Creating, then hiding a new toolbar button is the
dirty way i found to access utils.bindGlobalShortcut()
from Trilium API.
*/
api.addButtonToToolbar({
    title: 'Next Tab',
    shortcut: 'ctrl+pagedown',
    action: async function() {
        var e = $.Event("keydown");
        e.ctrlKey = true;
        e.which = 9; // # tab
        $("input").trigger(e);
    }
});
$('#toolbar-button-Next-Tab').hide()
```

This binds Ctrl+PageDown to go to previous tab.
But it's actually just an hidden `ButtonToolbar`, which triggers the original shortcut who goes to prevous tab.

It is probably a limitation of my javascript skills, but may be related to API limitations.
Anyway, if you have suggestions on how to achieve this in a clean way i'll be happy to see :)

**Comment by zadam** [comment]
Hi, I tried the script and it seems to work as intended. Of course it's a hack, but currently there isn't a better way to do this - although what I would change is `$("input").trigger(e);` which seems dangerous as it triggers this event on all inputs in DOM. I tried `$("#toolbar-button-Next-Tab").trigger(e);` instead and that would be a bit better.

But anyway, eventually this should not be necessary - configurable keyboard shortcuts are planned in relatively near future. Also as you say the API is limited at the moment - shortcut binding API should be created as well in the future.
